### PR TITLE
Add command to format Go code

### PR DESCRIPTION
**Pull request by tranngoclam**

### CODE REVIEW

File: Makefile
Comment (marcosnils) [inline Makefile:19]: nit: replace `command` with $(command)

**Review by marcosnils — COMMENTED** [review]
Thx for the contribution @tranngoclam! seems like linter is complaining about some changes. Mind checking those and re-submitting to review? thx

File: Makefile
Comment (tranngoclam) [reply Makefile]: @marcosnils: just updated, please help to review, thx

**Review by TomChv — APPROVED** [review]
LGTM, I think we should wait for @aluzzardi approval before merge.

**Review by gerhard — CHANGES_REQUESTED** [review]
Thank you for the first contribution @tranngoclam! 🙌🏻

This PR introduces two changes:
1. Add make target to format Go code
2. Run command and apply Go formatting changes

Both of those are good things, but there is one subtle implication: all contributors will be expected to run `make fmt` prior to committing.

In practice, this is unlikely to happen (people tend to forget), meaning that the utility of this change is limited from the start. Sure, we could add it to a checklist, but that will only add extra steps to contributing,  which will make it a less enjoyable experience. A better proposal would be to add this to our CI 😉

We use GitHub Actions today, and if your instinct is to add a workflow, let's talk about https://github.com/dagger/dagger/tree/v0.2.4/ci. The more targets we add to this `Makefile` (like your change proposes), the more difficult it will be to port them to Dagger. A better approach would be to:
- add `go fmt` & `gofumpt` to `pkg/universe.dagger.io/go`
- make it available via our `ci` dir
- integrate this into a GitHub Action that will fail if there is any diff (https://github.com/dagger/dagger/blob/v0.2.4/.github/workflows/lint.yml is a good starting point, but it is going in a wrong direction since it doesn't stand by one of our Dagger propositions - **an alternative to Make**)

I hope that sharing the above @tranngoclam will help you figure out the next steps towards a contribution that we can accept.

I am very excited to see you participate, thank you for taking the time! 🙌🏻

**Comment by helderco** [comment]
Yes, we're running CUE lint in CI, we should also run go fmt.

**Comment by dolanor** [comment]
I agree with all the decision behind the creation of `gofumpt`.
I think it will be a great addition to the `ci` Go steps.

**Comment by gerhard** [comment]
This was a promising contribution, thank you @tranngoclam for kicking it off.

In the absence of recent progress, I will assume that this dropped off your current list of priorities. That is OK, I know too well how that works 🙂 . In order to keep the PRs flowing, I will close this in order for us to keep a clean view of what changes are WIP and likely to land.

If you still want to contribute a subset of these changes, please open a new PR and reference this one. Thank you!

**Pull request closed without merging**
